commands/mr/create/mr_create.go

```diff
@@ -209,6 +204,20 @@ func NewCmdCreate(f *cmdutils.Factory) *cobra.Command {
 				}
 			}
 
+			if opts.TargetBranch == "" {
+				// Check if have a tracking branch (MergeRef) and use it as target branch
+				branchConfig := git.ReadBranchConfig(opts.SourceBranch)
+				if branchConfig.RemoteName != "" && branchConfig.MergeRef != "" {
```

Comment:
`git.ReadBranchConfig` is likely to return a nil struct if an error occurs. See

https://github.com/profclems/glab/blob/b2922a72dcf07f25a95046cdd3e9ff0afc2e1ff3/internal/git/git.go#L234-L240

 so it is important to check if `branchConfig` is not nil before accessing the fields to avoid panic

Reply:
I tested if it returns something that isn't `(git.BranchConfig{})`, hope that is enough